keras/callbacks_test.py

```diff
@@ -1869,6 +1869,51 @@ def set_weight_to_epoch(self, epoch):
         self.assertEqual(epochs_trained, 5)
         self.assertEqual(early_stop.model.get_weights(), 2)
 
+    def test_EarlyStopping_with_start_from_epoch(self):
+        with self.cached_session():
+            np.random.seed(1337)
+            (data, labels), _ = test_utils.get_test_data(
+                train_samples=TRAIN_SAMPLES,
+                test_samples=TEST_SAMPLES,
+                input_shape=(INPUT_DIM,),
+                num_classes=NUM_CLASSES,
+            )
+            labels = np_utils.to_categorical(labels)
+            model = test_utils.get_small_sequential_mlp(
+                num_hidden=NUM_HIDDEN,
+                num_classes=NUM_CLASSES,
+                input_dim=INPUT_DIM,
+            )
+            model.compile(
+                optimizer="sgd", loss="binary_crossentropy", metrics=["acc"]
+            )
+            start_from_epoch = 2
+            patience = 3
+            stopper = keras.callbacks.EarlyStopping(
+                monitor="acc",
+                patience=patience,
+                start_from_epoch=start_from_epoch,
+            )
+            history = model.fit(
+                data, labels, callbacks=[stopper], verbose=0, epochs=20
+            )
+            # Test 'patience' argument functions correctly when used
+            # in conjunction with 'start_from_epoch'.
+            assert len(history.epoch) >= patience + start_from_epoch
```

Comment:
Thanks @rchao for your comments.
regarding question 1 - the summation is a one way of checking that 
both existing "patience" mechanism is working as well as the new logic "start_from_epoch".
The "start_from_epoch" was created for the cases were indeed there's no improvement
in the model fit process for at least the amount of initial epochs designated by this "start_from_epoch" 
number. Afterwards, the existing "patience" mechanism should start counting and therefore 
in the worse-case the number of epochs should be at least more than this sum. I hope it makes sense. 
regarding question 2 - of-course I changed the asserts to self.assertX in the commit below.

Reply:
Thanks @inonbe for the explanation! For question 1, I'm still not fully following: imaging that this training of a model always increases the accuracy as epoch goes on, isn't it true that the early stopping callback will never attempt to stop it, since it's using `monitor="acc"`, and by definition the early stopping callback "Stop training when a monitored metric has stopped improving"? Please let me know if I'm missing anything.

Reply:
Hi Rick,
you're right when assuming that the accuracy keeps improving, but the purpose of this callback in general
and specially the argument i added are for the cases in which this is not the case.
I.e. when the accuracy is not improving and even degrading initially (regarding the "start_from_epoch")
or degrading after some period of training in the general case of the early stopping callback.
Since this test is relatively short in regard to number of epochs and the input/output data is totally random - 
the accuracy might indeed may randomly degrade occasionally.
I hope its more understood - thanks anyway for your efforts
Inon.